### PR TITLE
rust 1.72.1

### PR DESCRIPTION
**Pull request by chenrui333**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

- [x] `resource` blocks have been checked for updates.

### CODE REVIEW

**Comment by chenrui333** [comment]
cargo tags did not get marked yet, https://github.com/rust-lang/cargo/tags

**Comment by ZhongRuoyu** [comment]
Let's wait for #138176 to finish; I added several minor tweaks to `rust` in that PR. It will need a rerun due to `include-what-you-use` and a merge conflict in `castxml`, but that should be quick since most of the bottles from the current run will be reused. After that's merged, we can also see if we can switch `rust` to `llvm` 17 here.

**Comment by chenrui333** [comment]
also removing build from source label as this is more of a patch version bump (to save some CI bandwidth)

**Comment by ZhongRuoyu** [comment]
> cargo tags did not get marked yet, https://github.com/rust-lang/cargo/tags

It's the same as 1.72.0, so we don't need to update the `cargo` resource here.

**Comment by ZhongRuoyu** [comment]
Pushed some minor improvements. Though we still need a rebase after #138176.

File: Formula/r/rust.rb [hunk 1]
```diff
@@ -2,30 +2,19 @@ class Rust < Formula
...
-
-    # Fix for a compiler issue that may cause some builds to spin forever.
-    # Remove on 1.73.0 release.
-    # https://github.com/rust-lang/rust/pull/114948
-    # https://github.com/rust-lang/rust/issues/115297
-    # https://github.com/kpcyrd/sh4d0wup/issues/12
-    patch do
-      url "https://github.com/rust-lang/rust/commit/0f7f6b70617fbcda9f73755fa9b560bfb0a588eb.patch?full_index=1"
-      sha256 "549f446612bef10a1a06a967f61655b59c1b6895d762d764ca89caf65c114fe9"
-    end
```
Comment (ZhongRuoyu) [inline Formula/r/rust.rb:28]: This is now fixed, according to rust-lang/rust#115787.

**Comment by chenrui333** [comment]
should be good now.

**Comment by chenrui333** [comment]
cancelling the build, will do after llvm 17 bottle

**Comment by ZhongRuoyu** [comment]
Pushed an update to migrate to `llvm` 17, while we wait.

**Comment by ZhongRuoyu** [comment]
> Pushed an update to migrate to `llvm` 17, while we wait.

Tried that locally and bumped into rust-lang/rust#116020. We'll do that in 1.73.0 (the next minor version bump) instead.

**Review by ZhongRuoyu — APPROVED** [review]

**Pull request merged**